### ISSUE TITLE
New region lock tag system

### ISSUE THREAD

**Issue by Tecfan** [bartervg/barter.vg#208]
# What problem does this feature address?
Some games have weird region locks. Region lock tags are not always sufficient.

## Describe a solution
An alternative system. Slightly discussed here: https://discord.com/channels/329829406153375755/438406560222281738/797175045322571828

I have a feeling that @bartervg had some thoughts on this already, so mainly making this thread to document the idea/work officially.

**Comment by Luckz** [comment]
Can't even imagine a region lock tag system that would be specific enough to handle every possible random publisher idea. Equipping a tradable with hyper-specific lock tags has the same issue as being overly detailed with what app IDs are in a package: these things can change, and since Barter is not SteamDB (and is not permitted to scrape it in the most vulgar of ways), the Barter database cannot have package details updated through other means than manual labour.

**Comment by Tecfan** [comment]
I was thinking manual labor though, like the custom activation deadlines, you could have custom region locks and possibly parse a list from steamdb to your own tradable, and barter.vg could do whatever with this information.

For example:
Show the list of countries where it will or will not activate when hovering the Region lock trading tag.
Try to guess a match/not a match based on the information listed
Convert short country codes to full country names with or without flag icons

The tradable list on item pages would still only have a Region lock tag, but things could be shown if you hover or expand this tag. 

Alternatively, we could re-route the empty Region lock tag (as opposed to country specific region lock tags) to support subID's in the custom integer field, and have the Region lock tag then link to steamdb so that users can manually verify the weird region lock.

**Comment by bartervg** [comment]
> Slightly discussed here: https://discord.com/channels/329829406153375755/438406560222281738/797175045322571828 

This was a different issue: that region lock comment in an offer was causing the layout problem shown in 
![image](https://user-images.githubusercontent.com/28576132/105635135-6bccf480-5e2f-11eb-9cab-1df5b6c4af5f.png)

There was an alternative region lock system introduced in 2017 with this bundle for Endless Legend, but it was never widely implemented. It uses the region lock tag + ID, which corresponds to a list of included and excluded countries. As luckz explained, `manual labour`. If editors aren't willing to use it, what hope is there the average trader will? 

If you or others are willing to tolerate a difficult system in the name of full accuracy, I added the region lock editor button on the giveaway page. As with most abandoned features, support is limited. 
![image](https://user-images.githubusercontent.com/28576132/105635369-66bc7500-5e30-11eb-9f17-0198f05662ec.png)

**Comment by bartervg** [comment]
Is this a trying-to-recreate-SteamDB problem? Rather than creating bespoke region locks, wouldn't it be better to tag it with the subID? Although I did suggest and implement such a system, I underestimated the amount of work to maintain it. No one has even created region locks.

In the utopian version of the site, where packages are first-class citizens, the region lock information would be associated with the subID. Users wouldn't need to input 50 country codes as long as they got the subID correct.

**Comment by Revadike** [comment]
exactly

**Comment by arseny92** [comment]
> There was an alternative region lock system introduced in 2017 with this bundle for Endless Legend, but it was never widely implemented. It uses the region lock tag + ID, which corresponds to a list of included and excluded countries. As luckz explained, `manual labour`. If editors aren't willing to use it, what hope is there the average trader will?
> 
> If you or others are willing to tolerate a difficult system in the name of full accuracy, I added the region lock editor button on the giveaway page. As with most abandoned features, support is limited.

I completely forgot about this, and the Add Region Lock page lacks documentation about how and where to use the region lock ID after you add one, so likely that's why haven't used it to not potentially mess up things

Using just the subID tag without any other tags does not really encourage the user to go check there for any locks to see if his region is included or excluded, especially when viewing user's tradable collection such as mine where all items are linked with subIDs and one can get easily lost and some items have region *exclusions* instead of locks and using the `locked: old region set` or `region locked` tags does not help the user to distinguish exclusions: he may think the tradable is locked to the whole old region set and pass by. Most recent example of weird region lock where not all regions belonging to a region **set** (such as "CIS", "EU", "LATAM", "MEA") are listed, is on [Humble Bundle: Payday 2: The Ultimate Score Bundle](https://barter.vg/bundle/6463/). Does this mean the user needs to use a region lock ID on a tradable (with which tag?)? Do bundle editors need to also use said region lock ID on item tags on bundles? How does that shown in collection? *Is the region lock list accessible to a mortal user without extra permissions?* If no, how you expected this to be used by users in the first place? When there would be many region lock IDs and you're struggling to find the correct one to use, and you end up adding the same region set pair again, does it match to an existing set if one exists (like when you add relations to an existing Steam item through the Add Item form) and does not add a duplicate pair? Do the region lock form check field values to match (not as string, but region codes array, because the same region string can be in any order), not ID like the Add Item form does, to prevent duplicate region lock IDs?

What is this recent tedency to just close all these issues outright?

**Comment by bartervg** [comment]
> Using just the subID tag without any other tags does not really encourage the user to go check there for any locks 

True, but I think efforts should be devoted to upgrading the subID tag rather than the region locked tag.

>  how you expected this to be used by users in the first place? When there would be many region lock IDs and you're struggling to find the correct one to use

I think this is a good summary of  why the region lock ID system is a dead end.

> What is this recent tedency to just close all these issues outright?

Yes, this closure was premature since Tecfan's issue is broader than the region lock suggestion I offered. Use `subID` is not adequate yet.
